### ISSUE TITLE
Trust All Certs?

### ISSUE THREAD

**Issue by chadequate** [pivotal-cf/cloudfoundry-certificate-truster#12]
I believe I've implemented the truster correctly, but the errors I'm getting aren't any different than what I had prior. 

[ERR] [ERROR] ErrorPageFilter - Forwarding to error page from request [/api/v1/endpoint] due to exception [403 Forbidden] <org.springframework.web.client.HttpClientErrorException: 403 Forbidden>org.springframework.web.client.HttpClientErrorException: 403 Forbidden

To rule out configuration issues I'm wondering if there's a way to accept all certs; a wildcard maybe?

**Comment by scottfrederick** [comment]
@chadequate What's the use case you are trying to implement? This library is meant to serve a specific use case of trusting self-signed SSL certs from a Cloud Foundry component in apps running on Cloud Foundry. It is not meant to be a general-purpose SSL certificate handling library. 

There are better ways to handle certificates from non-root CAs in Cloud Foundry. If you are deploying CF with BOSH you can [configure trusted certificates](http://bosh.io/docs/trusted-certs.html) that will be placed on every VM that BOSH manages. This is [supported in Pivotal CF](https://docs.pivotal.io/pivotalcf/1-11/customizing/vsphere-config.html#security-config) also. The Java buildpack will [add these trusted certs to the JVM automatically](https://github.com/cloudfoundry/java-buildpack/blob/master/docs/framework-container_security_provider.md) when Java apps are run in CF. 

You can also fork the Java Buildpack and[ add certs directly to the buildpack](https://github.com/cloudfoundry/java-buildpack/blob/master/docs/jre-open_jdk_jre.md#custom-ca-certificates). 

Either of these solutions is preferable to having client applications add certificates directly by downloading them from the remote site. 

> I'm wondering if there's a way to accept all certs; a wildcard maybe?

You can configure the underlying HttpClient to trust all certs with configuration like what's described in various answers to this Stack Overflow question: https://stackoverflow.com/questions/2703161/how-to-ignore-ssl-certificate-errors-in-apache-httpclient-4-0. Of course, this defeats SSL security in your application and is a bad idea for production apps.

**Comment by chadequate** [comment]
I'm attempting to trust a cert issued from a company CA, served by an app hosted on Azure. As a developer I don't believe I have access to certificate management in Pivotal CF, so I was looking for a temporary way around this block. I stumbled on this tool today and thought I'd give it a shot, but based on your description it seems like it's probably not what I want (I've also tried the HttpClient method but I feel like PCF is still blocking me). I suppose I'll just have to wait for direction from PCF admins at my company, but thanks for the prompt response!
